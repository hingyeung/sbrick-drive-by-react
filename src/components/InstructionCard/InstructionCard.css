.instruction-card {
    /* need this bottom margin to maintain gap between cards that are not being dragged */
    /*margin: 0 0 8px 0;*/
    text-align: left;
    padding: 30px 0;
}

.instruction-card--is-dragging {
    box-shadow: 0 0 20px rgba(180, 180, 180, 1);
}

.instruction-card .instruction-card__icon {
    width: 25px;
    height: 25px;
    /*margin-left: 40px;*/
}

.instruction-card .instruction-card__label {
    /*margin-left: 40px;*/
    text-transform: uppercase;
}

.instruction-card__detail {
    padding: 0 15px 0 40px;
}