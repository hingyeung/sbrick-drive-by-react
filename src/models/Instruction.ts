export interface Instruction {
    displayName: string;
    id: string;
    index: number;
}
