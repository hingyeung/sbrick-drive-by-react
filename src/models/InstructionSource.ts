export interface InstructionSource {
    id: string;
    displayName: string;
    command: string;
}
