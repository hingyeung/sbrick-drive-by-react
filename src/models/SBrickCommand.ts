export enum SBrickCommand {
  forward,
  backward,
  left,
  right
}