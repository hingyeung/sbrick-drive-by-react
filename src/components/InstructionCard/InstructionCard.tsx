import * as React from 'react';
import { Draggable } from 'react-beautiful-dnd';
import { Instruction } from '../../models/Instruction';
// import './InstructionCard.css';

export interface Props {
  instruction: Instruction;
  index: number;
  className?: string;
}

export default (props: Props) => (
  <Draggable draggableId={props.instruction.id} index={props.index}>
    {(provided, snapshot) => (
      <div className={props.className + ' instruction-card'}>
        <div
          ref={provided.innerRef}
          className="instruction-card"
          style={{...provided.draggableProps.style as object}}
          {...provided.dragHandleProps}
        >
          {props.instruction.displayName}
        </div>
        {provided.placeholder}
      </div>
    )}
  </Draggable>
);