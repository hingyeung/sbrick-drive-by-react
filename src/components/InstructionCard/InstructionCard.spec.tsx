import * as renderer from 'react-test-renderer';
import * as React from 'react';
import { Instruction } from '../../models/Instruction';
// https://github.com/facebook/jest/issues/936#issuecomment-214556122
// rbd is used to mock 'react-beautiful-dnd'
import * as rbd from 'react-beautiful-dnd';
import { DraggableProvided } from 'react-beautiful-dnd';
import { SBrickCommand } from '../../models/SBrickCommand';
import InstructionCard from './InstructionCard';

jest.mock('react-beautiful-dnd');

const INSTRUCTION: Instruction = {
  id: 'instructionId',
  displayName: 'displayName',
  sBrickCommand: SBrickCommand.forward
};

describe('InstructionCard', function () {
  beforeEach(() => {
    const draggableProvided: DraggableProvided = {
      innerRef: () => {
        return;
      },
      draggableProps: {
        style: null,
        'data-react-beautiful-dnd-draggable': ''
      },
      dragHandleProps: null
    };
    // even though tslint is complaining about "attempted to assign to readonly property",
    // it works, and I don't know how to do it any other way.
    // @ts-ignore
    rbd.Draggable = ({children}: any) => (<div>{children(draggableProvided)}</div>);
  });

  it('should render properly', () => {
    const tree = renderer
      .create(
        <InstructionCard index={0} instruction={INSTRUCTION} icon="icon"/>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});