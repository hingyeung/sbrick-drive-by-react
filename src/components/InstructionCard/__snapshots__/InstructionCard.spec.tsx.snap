// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionCard should render properly 1`] = `
<Component
  className="extraClass"
  icon="icon"
  index={0}
  instruction={
    Object {
      "displayName": "displayName",
      "id": "instructionId",
      "sBrickCommand": 0,
    }
  }
>
  <Component
    draggableId="instructionId"
    index={0}
  >
    <div>
      <div
        className="draggable-container"
      >
        <div
          className="instruction-card  extraClass"
          style={Object {}}
        >
          <img
            className="instruction-card__icon"
            src="icon"
          />
          <span
            className="instruction-card__label"
          >
            displayName
          </span>
        </div>
      </div>
    </div>
  </Component>
</Component>
`;

exports[`InstructionCard should render properly when dragging 1`] = `
<Component
  className="extraClass"
  icon="icon"
  index={0}
  instruction={
    Object {
      "displayName": "displayName",
      "id": "instructionId",
      "sBrickCommand": 0,
    }
  }
>
  <Component
    draggableId="instructionId"
    index={0}
  >
    <div>
      <div
        className="draggable-container"
      >
        <div
          className="instruction-card instruction-card--is-dragging extraClass extraClass--is-dragging"
          style={Object {}}
        >
          <img
            className="instruction-card__icon"
            src="icon"
          />
          <span
            className="instruction-card__label"
          >
            displayName
          </span>
        </div>
      </div>
    </div>
  </Component>
</Component>
`;
